src/VaultV2.sol

```diff
@@ -398,21 +397,33 @@ contract VaultV2 is IVaultV2 {
         emit EventsLib.RemoveAdapter(account);
     }
 
-    /// @dev This function requires great caution because it can irreversibly disable submit for a selector.
-    /// @dev Existing pending operations submitted before increasing a timelock can still be executed at the initial
-    /// executableAt.
+    /// @dev Existing pending operations submitted before increasing a timelock can still be executed. The increase of a
+    /// timelock only increases the timelock on future submissions.
     function increaseTimelock(bytes4 selector, uint256 newDuration) external {
         timelocked();
         require(selector != IVaultV2.decreaseTimelock.selector, ErrorsLib.TimelockIsAutomatic());
+        require(newDuration <= TIMELOCK_CAP, ErrorsLib.TimelockDurationTooHigh());
         require(newDuration >= timelock[selector], ErrorsLib.TimelockNotIncreasing());
 
         timelock[selector] = newDuration;
         emit EventsLib.IncreaseTimelock(selector, newDuration);
     }
 
+    /// @dev Irreversibly disable submit for a selector.
+    /// @dev Be particularly careful as this action is not reversible.
+    /// @dev Existing timelocked operations submitted before abdicating the selector can still be executed. The
+    /// abdication of a selector only prevents future operations to be submitted.
+    function abdicateSubmit(bytes4 selector) external {
```

Comment:
I wasn't thinking about separated function, but works too

Reply:
I started having an `increaseTimelock` that checks `<= 1 year || == uint.max` but it really called for a separate function. In code but also you can have a bigger timelock for abdicate.